exp/controllers/alias.go

```diff
@@ -24,7 +24,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 
 	"sigs.k8s.io/cluster-api/controllers/clustercache"
-	machinepool "sigs.k8s.io/cluster-api/exp/internal/controllers"
+	"sigs.k8s.io/cluster-api/exp/internal/controllers/machinepool"
```

Comment:
Am I wrong or this change is a little bit orthogonal to this PR (you can also just add the owner file to sigs.k8s.io/cluster-api/exp/internal/controllers, there is nothing else there)
Context: I would prefer to not add further nesting on the exp folders, when we sort of agreed to move MP folders out of exp folder when talking about graduation

Same for CAPD

Reply:
Yup

Reply:
If we're okay with that

Reply:
CAPD should be test/infrastructure/docker/internal/controllers/dockermachinepool/ (

Reply:
I took @sbueringer's suggestion

> I think one thing we can do is move some code around to move machinepool and CAPD DockerMachinePool controller code into machinepool directories and add Owner files there

Reply:
I let Fabrizio decide

Reply:
CAPD doesn't have yet a folder for each controller, let's move the content of test/infrastructure/docker/exp/internal/controllers to test/infrastructure/docker/internal/controllers and rename files as dockermachinepool as suggested by Jack (we can add Matt and Richard into cluster-api-test-reviewers too)

Also, what about the webhooks folders

Reply:
Not sure if we need that fine-granular owner ship in webhooks too. We have the same question for APIs and then the next question is: what if any of that code uses shared code.

I think for now I would mostly stick to the MP controller code

In general OWNERS might also allow ownership on a per-file level. But IIRC this broke a few years ago and I'm not sure if that's fixed already

Reply:
I think webhooks should ideally live right next to the api type definition, and be an inherited maintenance responsibility as part of maintaining the API itself. I assume we don't want to let perfection be the enemy of the good here, and not worry about refactoring the way that capi organizes webhooks in a separate source directory structure from the API types.

Reply:
> and not worry about refactoring the way that capi organizes webhooks in a separate source directory structure from the API types.

We spend a lot of time refactoring the webhooks away from the API: https://github.com/kubernetes-sigs/cluster-api/issues/9011

Reply:
even more reason not to do anything :)

Reply:
Got it. Let's also move the mp webhook code to the other webhooks but not add an owner file for it